### PR TITLE
Test operatorhub submission with quay.io repo

### PR DESCRIPTION
**Pull request by galderz**
This is an early preview for testing an operatorhub.io submission on minikube. Instructions need to still be added to show how to push submissions to quay.io. Right now it uses my own quay.io repo to which I've been pushing pre-releases.

* Pushed an operatorhub.io submission to own quay.io.
* Used that submission to test it on minikube.

Sending PR early to get some feedback. TODOs:

- [x] Make VM driver configurable (currently uses `virtualbox`)
- [x] Test with operator 0.2.1 once that's available.

### CODE REVIEW

**Comment by galderz** [comment]
This is now ready for review. It still uses operator OLM releases in my own quay.io repo, but that's ok for now. I'll push some releases OLM to infinispan quay.io repo. Operatorhub releases will come from their own quay.io repos, so our infinispan quay.io repo for OLM stuff should mostly be for testing.

File: operatorhub/minikube/Makefile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+install-operator:
+	./build/install-operator.sh ${NAMESPACE}
+
+test:
+	./build/test.sh ${NAMESPACE}
+
+delete:
+	./build/delete.sh ${PROFILE}
+
+all: config start checkout-olm install-olm checkout-marketplace install-marketplace install-operator test
```
Comment (rigazilla) [inline operatorhub/minikube/Makefile]: `all` can be the default target if moved to the top

File: operatorhub/minikube/Makefile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+PROG  := infinispan-operatorhub-minikube
+
+.PHONY: config clean start test help
+
+config:
+	./build/config.sh ${PROFILE} ${VMDRIVER}
+
+clean:
+	./build/clean.s ${NAMESPACE} ${VERSION}
```
Comment (rigazilla) [inline operatorhub/minikube/Makefile]: `h` missing. clean.sh

File: README.md [hunk 3]
```diff
@@ -139,6 +139,117 @@ Alternatively, pass `KUBECONFIG` to specify cluster access:
...
+
+* In the automatic test, the operator is added to the operator hub,
+then it's installed in the defined namespace,
+and tested all in one step:
+`make all`
+
+* In the manual test, `make install-operatorsource` is called to add the operator to the operator hub.
+Installing the operator into the desired namespace is done via the Operator Hub user interface in the OpenShift console.
+Finally, the operator can be tested by calling `make test`.
+
```
Comment (rigazilla) [inline README.md]: Is `make clean` to cleanup? Or I need `make delete` also?

File: operatorhub/minikube/Makefile [hunk 4]
```diff
@@ -0,0 +1,49 @@
+TMP_DIR ?= /tmp
+PROFILE ?= operatorhub.io
+VMDRIVER ?= virtualbox
+NAMESPACE ?= local-operators
+VERSION ?= 0.2.1
+
+PROG  := infinispan-operatorhub-minikube
+
+.PHONY: config clean start test help
```
Comment (rigazilla) [inline operatorhub/minikube/Makefile]: All the targets would seem to be phony.
There's no `help` target

File: README.md (hunk 3 above)
Comment (galderz) [reply README.md]: Clean just removes added resources to Kubernetes. Delete removes the VM altogether. I'll adjust the README.

**Comment by galderz** [comment]
@rigazilla Fixed your comments

**Comment by galderz** [comment]
@rigazilla This is ready now

**Comment by rigazilla** [comment]
merged, thanks @galderz !

**Pull request merged**